.mergify.yml

```diff
@@ -0,0 +1,23 @@
+queue_rules:
+  # This rule is for https://docs.mergify.com/commands/queue/
+  # and can be triggered with: @mergifyio queue
+  - name: default
+    merge_conditions:
+      # all github action checks in this list are required to merge a pull request
+      - check-success=Attributes
+      - check-success=Check
+      - check-success=Outpaths (aarch64-darwin)
+      - check-success=Outpaths (aarch64-linux)
+      - check-success=Outpaths (x86_64-darwin)
+      - check-success=Outpaths (x86_64-linux)
+      - check-success=Process
+      - check-success=Request
+      - check-success=Tag
+      - check-success=editorconfig-check
+      - check-success=label-pr
+      - check-success=nix-files-parseable-check
+      - check-success=nixfmt-check
+      - check-success=nixpkgs-vet
```

Comment:
By-name and nixfmt checks should not be mandatory, and can be allowed to fail, I believe.

Reply:
This is for automatic merging, should we not prefer a human looking at the pull request if `nixfmt` or the `by-name` check fails? cc @infinisil 
Afaik `nixfmt` only triggers if the file was formatted before or if it is a new file.

Reply:
Nixfmt is probably fine, but I have seen some cases where nixpkgs-vet fails (like https://github.com/NixOS/nixpkgs/pull/332242)

A human reviewer/commiter is still needed to add the PR to the merge queue.

Reply:
> Nixfmt is probably fine, but I have seen some cases where nixpkgs-vet fails (like #332242)

I believe at least that is a correct one. `firedragon` can and should be put in `by-name`.

Reply:
> I believe at least that is a correct one. `firedragon` can and should be put in `by-name`.

https://github.com/NixOS/nixpkgs/pull/332242#discussion_r1829448213

Reply:
I'm pretty sure that is a valid case and the error previously reported is not related to the position, but e.g. rebasing to master. The original diff is garbage collected by GitHub so I cannot check it again. When this happens, we can at least check:

1. Can you verify the `outPath` of the package remain the same, when inside `by-name` and outside?
2. Do you have a dynamic reference (generated by function) to files outside the path?

Rather than believe in what people casually says (I have encountered a lot of prevarication replies tbh), we should find out why. If we just randomly throw confusing code into the codebase, we will make it more difficult for those who maintain the code in the future. And if this situation truly reflects a problem with our underlying code, we should at least report it promptly.

Reply:
Ok, I have no problem with these tests being a requirement.